java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java

```diff
@@ -243,6 +243,28 @@ public void visitMethodInvocation(MethodInvocationTree tree) {
     registerType(tree, returnType);
     if(resolution != null) {
       inferArgumentTypes(argTypes, resolution);
+      List<JavaType> parameterTypes = getParameterTypes(tree.arguments());
+      if(parameterTypes != argTypes) {
+        IdentifierTree identifier;
+        if (methodSelect.is(Tree.Kind.MEMBER_SELECT)) {
+          MemberSelectExpressionTree mset = (MemberSelectExpressionTree) methodSelect;
+          JavaType type = getType(mset.expression());
+          if(type.isTagged(JavaType.DEFERRED)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS134)